File: Esercitazione02_2021/java-aux-170320/MaxGap.java
public class MaxGap {

    public static int maxGap(int[] array, int start, int end ){
      if(end - start == 1 ) //size = 2
        return array[end]-array[start];
      else if(end-start == 2 ) //size = 3
        return Math.max(array[start+1]-array[start], array[end]-array[start+1]);
      else{
        return Math.max(maxGap(array, start, (end+start)/2), maxGap(array,(end+start)/2, end));
      }
    }
    
}

File: ESAMI/DOCS_NOTES/Fondamenti di Informatica II/AlgoritmiEStruttureDati-master/AlgoritmiEStruttureDati-master/Algoritmi - Esami/ALGORITMI/RICORSIONE/max_gap.c
#include <assert.h>
#include <stdio.h>

#include "max_gap.h"

int maxGap(int * array, int start, int end) {
	// TODO: Da completare
	if(end-start == 2){
		return array[start+1] - array[start];
	}
	int n = (end-start)/2;
	int l= maxGap(array, start, n+start+1);
	int r=maxGap(array,n+start,end);
	if(l>r) return l;
	else return r;
}
